src/core/mount.c

```diff
@@ -1439,7 +1439,10 @@ static int mount_deserialize_item(Unit *u, const char *key, const char *value, F
         } else if (streq(key, "control-pid")) {
 
                 if (!pidref_is_set(&m->control_pid))
-                        (void) deserialize_pidref(fds, value, &m->control_pid);
+                        (void) deserialize_pidref(fds, value, !u->deserialize_got_pidfd, &m->control_pid);
+
+                if (m->control_pid.fd >= 0)
+                        u->deserialize_got_pidfd = true;
```

Comment:
How does this solve the problem? Like I said, if the previous process died, nothing is attached to `control_pid`...